File: valid-parentheses.py
def valid_parentheses(string):
  balance = 0
  for item in string:
    if item == "(":
      balance = balance + 1
    if item == ")":
      balance = balance - 1
    if balance < 0:
      return False
  if balance != 0:
    return False
  return True

File: valid-parentheses.js
const validParentheses = parens => {
	let balance = 0;
	for (let i = 0; i < parens.length; i++) {
		if (parens[i] === '(') { balance ++; }
		if (parens[i] ===')') { balance --; }
		if (balance < 0) { return false; }
	}
	if (balance !== 0) { return false; }
	return true;
};